packages/core-app-api/src/app/AppManager.tsx

```diff
@@ -341,6 +344,19 @@ export class AppManager implements BackstageApp {
         const featureFlagsApi = this.getApiHolder().get(featureFlagsApiRef)!;
 
         if (featureFlagsApi) {
+          const configFlags =
+            this.configApi?.getOptionalConfigArray('app.featureFlags');
+          if (configFlags && configFlags.length > 0) {
+            for (const configFlag of configFlags) {
+              featureFlagsApi.registerFlag({
+                name: configFlag.getString('name'),
+                description: configFlag.getOptionalString('description'),
+              });
+            }
+          }
+          for (const flag of this.featureFlags) {
```

Comment:
I am not sure what flags are being registered here. Or is it that you want to make the `AppManager` more flexible and also allow for feature flags to passed in directly when creating the app instance? I wonder if that is needed if you can already pass it through the `app-config.yaml`.

Reply:
Yeah, flexibility point of view, and the user can add the flags also dynamically if necessary.

Reply:
On second thought, it might be just too much. I will remove the possibility for passing flags in `createApp` functionality